### PR TITLE
Fix/logging

### PR DESCRIPTION
**Pull request by martinfalk**
* added `LogFilter` for filtering based on `LogVerbosity`
* custom logger can be supplied to Deserializer

### CODE REVIEW

File: include/inviwo/core/util/logfilter.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+namespace inviwo {
+
+/**
+ * \class LogFilter
+ * \brief filters log messages based on the verbosity level. Messages below the selected verbosity
+ * will be ignored.
+ */
+class IVW_CORE_API LogFilter : public Logger {
+public:
+    LogFilter(Logger* logger);
```
Comment (petersteneteg) [inline include/inviwo/core/util/logfilter.h]: Could default to logcentral and all messages

File: include/inviwo/core/util/logfilter.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    LogFilter(Logger* logger, LogVerbosity verbosity);
+    virtual ~LogFilter() = default;
+
+    void setVerbosity(LogVerbosity verbosity);
+    LogVerbosity getVerbosity();
+
```
Comment (petersteneteg) [inline include/inviwo/core/util/logfilter.h:53]: Add get/set for the logger

File: include/inviwo/core/io/serialization/deserializer.h [hunk 3]
```diff
@@ -52,7 +52,7 @@ class FactoryBase;
 template <typename T, typename K>
 class ContainerWrapper;
 
-class IVW_CORE_API Deserializer : public SerializeBase {
+class IVW_CORE_API Deserializer : public SerializeBase, public Logger {
```
Comment (petersteneteg) [inline include/inviwo/core/io/serialization/deserializer.h]: Derive from LogFilter, then you dont need to add any impl here...

File: src/core/util/logfilter.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+LogFilter::LogFilter(Logger* logger, LogVerbosity verbosity)
+    : logVerbosity_{verbosity}, logger_{logger} {}
+
+void LogFilter::setVerbosity(LogVerbosity verbosity) { logVerbosity_ = verbosity; }
+
+LogVerbosity LogFilter::getVerbosity() { return logVerbosity_; }
+
+void LogFilter::log(std::string logSource, LogLevel level, LogAudience audience,
+                    const char* file, const char* function, int line, std::string msg) {
+    if (level >= logVerbosity_) {
```
Comment (petersteneteg) [inline src/core/util/logfilter.cpp:49]: Move string?

**Review by petersteneteg — APPROVED** [review]
Cool nice generalization!

File: include/inviwo/core/io/serialization/deserializer.h (hunk 3 above)
Comment (martinfalk) [reply include/inviwo/core/io/serialization/deserializer.h]: good point!

**Pull request merged**